examples/usecases/retrieval-with-hyperparameter-optimization.ipynb

```diff
@@ -0,0 +1,1405 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #16.        customer_ids_preds = customer_id_mapping.to_pandas().customer_id.iloc[1:].values
</pre><p>why do we need a func for the <code>hit_rate_at_100?</code> is that a required set up by Optuna?  </p><p> </p><p>If not,  cant we use our own metrics class? we already defined <code>recall_at</code> function https://github.com/NVIDIA-Merlin/models/blob/main/merlin/models/tf/metrics/topk.py#L48, so cant we use recall@100? (hit rate and recall should be same).</p><p> </p><p>can we not do like below?</p><p> </p><p><code> metrics = [ mm.RecallAt(100)]</code></p><p><code> model.compile(optimizer, loss=loss, metrics=metrics)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/786/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Reply:
Yes, you are right. I believe this is needed by Optuna. We need a function that will transform the model into a `top_k_recommender` and that will output a hit count calculated on a per-customer basis (using customer purchases in the validation week and 100 top predictions for a given customer) hence I define this function, to pull together all that functionality.

Reply:
<p>Can't you just use model.evaluate() for that retrieval model instead of using <code>model.to_top_k_recommender()</code> ?</p><p>For example</p><p> </p><pre class="ql-syntax" spellcheck="false">eval_metrics = model.evaluate(
        eval_ds,
        batch_size=args.eval_batch_size,
        return_dict=True,
)
</pre><p>Or do you have some specific requirement for your evaluation?</p>

Reply:
Thank you very much, Gabriel, for pointing me to this PR 🙏From what I understand, this code is still a month or two away from being merged? Today on the example call I believe we agreed to trying to complete this PR as soon as we can and flagging it for an update when the PR you mentioned goes through as well.